src/twisted/application/app.py

```diff
@@ -702,4 +703,7 @@ def _exitWithSignal(sig):
     @type sig:  C{int}
     """
```

Comment:
Python 3.8 supports settings the exit code to -2 and running all the finalization normally with:

```suggestion
    """
    if sig == signal.SIGINT and sys.version_info > (3, 8):
        raise KeyboardInterrupt
```

but it's broken https://bugs.python.org/issue41602

Reply:
@altendky https://bugs.python.org/issue41602 is fixed upstream now and waiting for backport